ocaml/utils/language_extension.mli

```diff
@@ -103,11 +103,27 @@ val with_disabled : 'a t -> (unit -> unit) -> unit
     way. *)
 val restrict_to_erasable_extensions : unit -> unit
 
-(** Permanently ban all extensions; used for [-disable-all-extensions] to
-    ensure that some code is 100% extension-free.  When called, disables any
+(** Permanently ban all extensions; used for [-disable-all-extensions] to ensure
+    that some code is 100% extension-free.  When called, disables any
     currently-enabled extensions, including the defaults.  Causes any future
-    uses of [set ~enabled:true], [enable], and their [with_] variants to
-    raise; also causes any future uses of [only_erasable_extensions] to raise.
+    uses of [set ~enabled:true], [enable], and their [with_] variants to raise;
+    also causes any future uses of [restrict_to_erasable_extensions] to raise.
     The [is_enabled] function will still work, it will just always return
     [false].*)
 val disallow_extensions : unit -> unit
+
+(**/**)
+
+(** Special functionality that can only be used in "pprintast.ml" *)
+module For_pprintast : sig
+  (** A fully-polymorphic function that can wrap printers from "pprintast.ml" *)
+  type printer_exporter =
+    { export_printer :
+        'a. (Format.formatter -> 'a -> unit) -> (Format.formatter -> 'a -> unit)
+    }
+
+  (** Wrap a printer from "pprintast.ml" so that it will unconditionally print
+      Jane syntax, and will not raise an exception.  Raises if called more than
+      once ever. *)
+  val make_printer_exporter : unit -> printer_exporter
+end
```

Comment:
I agree that it's not _more_ concerning than many things, but the pattern is very cheap -- one `ref` and one `if`.  I don't want to take worse approaches just because we haven't used better ones in the past (see: the compiler _gradually_ growing labeled arguments), although I understand there's also disagreement about whether this is actually better!

`with_maximally_enabled` is what I went to write first, but it would have to make the safety checks; `enable` and `with_enabled` both make the safety checks, and I wouldn't want the analogy to differ.  So then I went to put it in `For_pprintast`, and then realized that it was cheap to get an actual guarantee as I did so.  It wouldn't be hard to expose `with_maximally_enabled_no_matter_what` or whatever from `For_pprintast` instead, but that's how we got here.

Also in re. libraries using `-disable-all-extensions` / `-only-erasable-extensions`: We set that for public release, so it has a very particular important use case despite being infrequent.

Reply:
I don't think the implementation is overcomplicated. I think the interface might be. (I see it demands 3 comments in the mli and 1 at its use site.)

The most convincing part of your argument is the fact that it would be confusing for `enable_maximal` to do safety checks and `with_maximally_enabled` to not. I suppose I wouldn't mind another function `with_safety_checks_disabled` but that seems to be encroaching on "overcomplicated" territory again.

OK, I think this is fine. I have some comments on names but I'll leave them in another review.

Reply:
FWIW, I agree that this is complicated; if it weren't for the disanalogy, I would have just implemented `with_maximally_enabled`, as I said.  Fundamentally, perhaps all of this is actually an argument for moving the extension checks into the parser, at which point this would all be moot.  But that's a bigger change.